test/framework/cloudstack.go

```diff
@@ -339,3 +351,101 @@ func (c *CloudStack) WithRedhatVersion(version anywherev1.KubernetesVersion) api
 		return nil
 	}
 }
+
+func (c *CloudStack) getDevRelease() *releasev1.EksARelease {
+	c.t.Helper()
+	if c.devRelease == nil {
+		latestRelease, err := getLatestDevRelease()
+		if err != nil {
+			c.t.Fatal(err)
+		}
+		c.devRelease = latestRelease
+	}
+
+	return c.devRelease
+}
+
+func (c *CloudStack) templateForDevRelease(osFamily anywherev1.OSFamily, kubeVersion anywherev1.KubernetesVersion) string {
+	c.t.Helper()
+	return c.templateForRelease(osFamily, c.getDevRelease(), kubeVersion)
+}
+
+func envVarForCloudStackTemplate(osFamily, eksDName string) string {
+	return fmt.Sprintf("T_CLOUDSTACK_TEMPLATE_%s_%s", strings.ToUpper(osFamily), strings.ToUpper(strings.ReplaceAll(eksDName, "-", "_")))
+}
+
+func defaultNameForCloudStackTemplate(osFamily, eksDName string) string {
+	return filepath.Join(fmt.Sprintf("%s-%s", strings.ToLower(eksDName), strings.ToLower(osFamily)))
+}
+
+func defaultEnvVarForCloudStackTemplate(osFamily string, kubeVersion anywherev1.KubernetesVersion) string {
+	return fmt.Sprintf("T_CLOUDSTACK_TEMPLATE_%s_%s", strings.ToUpper(osFamily), strings.ReplaceAll(string(kubeVersion), ".", "_"))
+}
+
+// templateForRelease tries to find a suitable template for a particular eks-a release, k8s version and OS family.
+// It follows these steps:
+//
+// 1. Look for explicit configuration through an env var: "T_CLOUDSTACK_TEMPLATE_{osFamily}_{eks-d version}" eg. T_CLOUDSTACK_TEMPLATE_REDHAT_KUBERNETES_1_23_EKS_22
+// This should be used for explicit configuration, mostly in local development for overrides.
+//
+// 2. If not present, look for a template if the default templates: "{eks-d version}-{osFamily}" eg. kubernetes-1-23-eks-22-redhat
+// This is what should be used most of the time in CI, the explicit configuration is not present but the right template has already been
+// imported to cloudstack.
+//
+// 3. If the template doesn't exist, default to the value of the default template env vars: eg. "T_CLOUDSTACK_TEMPLATE_REDHAT_1_23".
+// This is a catch all condition. Mostly for edge cases where the bundle has been updated with a new eks-d version, but the
+// the new template hasn't been imported yet. It also preserves backwards compatibility.
+func (c *CloudStack) templateForRelease(osFamily anywherev1.OSFamily, release *releasev1.EksARelease, kubeVersion anywherev1.KubernetesVersion) string {
```

Comment:
this method seems like it's a 1 to 1 with the vsphere logic
What do you think about trying to move it to a shared piece of logic?